src/sql/parts/jobManagement/views/jobsView.component.ts

```diff
@@ -374,4 +398,51 @@ export class JobsViewComponent implements AfterContentChecked {
 			});
 		}
 	}
+
+	private createJobChart(jobId: string, jobHistories: sqlops.AgentJobHistoryInfo[]): void {
+		let chartHeights = this.getChartHeights(jobHistories);
+		for (let i = 0; i < jobHistories.length; i++) {
+			let runGraph = $(`table#${jobId}.jobprevruns > tbody > tr > td > div.bar${i+1}`);
+			if (jobHistories && jobHistories.length > 0) {
+				runGraph.css('height', chartHeights[i]);
+				let bgColor = jobHistories[i].runStatus === 0 ? 'red' : 'green';
+				runGraph.css('background', bgColor);
+				runGraph.hover((e) => {
+					let currentTarget = e.currentTarget;
+					currentTarget.title = jobHistories[i].runDuration;
+				});
+			} else {
+				runGraph.css('height', '5px');
+				runGraph.css('background', 'red');
+				runGraph.hover((e) => {
+					let currentTarget = e.currentTarget;
+					currentTarget.title = 'Job not run.';
+				});
+			}
+		}
+	}
+
+	// chart height normalization logic
+	private getChartHeights(jobHistories: sqlops.AgentJobHistoryInfo[]): string[] {
```

Comment:
we should try to add unit tests for logic that doesn't have hard UI dependencies.